### ISSUE TITLE
UseDataClass - exclude classes with interfaces

### ISSUE THREAD

**Issue by imanushin** [detekt/detekt#2790]
## Expected Behavior
Class below shouldn't be data class

## Current Behavior
Class below is data class from detekt prospective

## Context
```
interface I1
interface I2

class Container(
     val i1: I1,
     val i2: I2
)
```

Class above is just container which walks between methods. It has interfaces only, so in the most cases the toString/equality methods are useless here.

Another example - context classes, which are used as service locators.

**Comment by cortinico** [comment]
Thanks for your this report @imanushin 

From my point of view, this looks like a special scenario that you should suppress locally in your code (nit: you could also define a custom annotation and use the `excludeAnnotatedClasses` config of `UseDataClass` to exclude such classes).

To follow up on your sample, this class:

```kotlin
class Mailbox(
    val inbox : List<String>,
    val outgoing : Queue<String>
)
```
that contains only interfaces seems like a good candidate for a `data class`.

Leaving open to hear others' opinion.

**Comment by schalkms** [comment]
I agree with Nico here. It’s hard to find a rule that decides, whether this is a candidate or not.

**Comment by BraisGabin** [comment]
I think that there's very little that we can do here. I think that the custom annotation is the best option, as @cortinico say.

If you have a lot of this classes other option is to disable this rule all together. Other option is to disable it. I don't think that this rule could fit every project. And there is not point to force it.

**Comment by schalkms** [comment]
Closed due to the reasons mentioned.

**Comment by imanushin** [comment]
@schalkms, @BraisGabin , agree, I see two ways here: close issue or create AI to split service interfaces from dto interfaces. It is reasonable just to close this issue, to avoid overcomplicated logic for this rule.

**Comment by schalkms** [comment]
If you can come up with something reasonable, we can discuss about merging it. For now, I don’t think it’s worth to put more effort into it also considering the number of items with higher priority.
